hoodie-client/src/main/java/com/uber/hoodie/config/HoodieIndexConfig.java

```diff
@@ -54,8 +54,35 @@
   public static final String HBASE_ZKPORT_PROP = "hoodie.index.hbase.zkport";
   public static final String HBASE_TABLENAME_PROP = "hoodie.index.hbase.table";
   public static final String HBASE_GET_BATCH_SIZE_PROP = "hoodie.index.hbase.get.batch.size";
+  // Note that if HBASE_PUT_BATCH_SIZE_AUTO_COMPUTE_PROP is set to
+  // true, this batch size will not be honored for HBase Puts
   public static final String HBASE_PUT_BATCH_SIZE_PROP = "hoodie.index.hbase.put.batch.size";
-  public static final String DEFAULT_HBASE_BATCH_SIZE = "100";
+  // Property to set to enable auto computation of put batch size
+  public static final String HBASE_PUT_BATCH_SIZE_AUTO_COMPUTE_PROP = "hoodie.index.hbase.put.batch.size.autocompute";
+  public static final String DEFAULT_HBASE_PUT_BATCH_SIZE_AUTO_COMPUTE = "false";
+  // Property to set the fraction of the global share of QPS that should be allocated
+  // to this job. Let's say there are 3 jobs which have input size in terms of number
+  // of rows required for HbaseIndexing as x, 2x, 3x respectively. Then this fraction for
+  // the jobs would be (0.17) 1/6, 0.33 (2/6) and 0.5 (3/6) respectively.
+  public static final String HBASE_QPS_FRACTION_PROP = "hoodie.index.hbase.qps.fraction";
+  public static final String HBASE_SLEEP_MS_PUT_BATCH_PROP = "hoodie.index.hbase.sleep.ms.for.put.batch";
+  public static final String HBASE_SLEEP_MS_GET_BATCH_PROP = "hoodie.index.hbase.sleep.ms.for.get.batch";
+  // Property to set maximum QPS allowed per Region Server. This should
+  // be same across various jobs. This is intended to limit the aggregate
+  // QPS generated across various jobs to an Hbase Region Server.
+  public static String HBASE_MAX_QPS_PER_REGION_SERVER_PROP = "hoodie.index.hbase.max.qps.per.region.server";
+  // Default batch size, used only for Get, but computed for Put
+  public static final int HBASE_DEFAULT_SLEEP_MS = 100;
+  // Default sleep duration between hbase queries (HBase Get or Put)
+  // Note that if HBASE_PUT_BATCH_SIZE_AUTO_COMPUTE_PROP is set to
+  // true, this batch size will not be honored for HBase Puts
+  public static final int DEFAULT_HBASE_BATCH_SIZE = 100;
+  // From benchmarking, Region Servers have been able to
+  // handle > 1 Mil puts/min, or 16667 puts/second
+  public static final int HBASE_DEFAULT_MAX_QPS_PER_REGION_SERVER = 16667;
```

Comment:
Is this number for any generic region server ? Does it depend on what kind of deployment there is, what hardware is used to run the cluster etc ? This DEFAULT may not make sense for a general config.

Reply:
Done